libs/shared/typescript/charts-ngx/src/lib/boxplot/boxplot.directive.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { Component } from '@angular/core';
+import {
+  BoxplotProps,
+  dynamicBoxplotPoints,
+  staticBoxplotPoints,
+  staticBoxplotSummaries,
+} from '@sagebionetworks/shared/charts';
+import { render, screen } from '@testing-library/angular';
+import { BoxplotDirective } from './boxplot.directive';
+
+const renderTestComponent = async (props: BoxplotProps) => {
+  @Component({
+    template: `<div
+      sageBoxplot
+      [points]="points"
+      [summaries]="summaries"
+      [title]="title"
+      [xAxisTitle]="xAxisTitle"
+      [yAxisTitle]="yAxisTitle"
+      [yAxisMin]="yAxisMin"
+      [yAxisMax]="yAxisMax"
+      [xAxisCategoryToTooltipText]="xAxisCategoryToTooltipText"
+      [pointTooltipFormatter]="pointTooltipFormatter"
+    ></div>`,
+  })
+  class TestComponent {
+    points = props.points;
+    summaries = props.summaries;
+    title = props.title;
+    xAxisTitle = props.xAxisTitle;
+    yAxisTitle = props.yAxisTitle;
+    yAxisMin = props.yAxisMin;
+    yAxisMax = props.yAxisMax;
+    xAxisCategoryToTooltipText = props.xAxisCategoryToTooltipText;
+    pointTooltipFormatter = props.pointTooltipFormatter;
+  }
+
+  return await render(TestComponent, {
+    imports: [BoxplotDirective],
+  });
+};
+
+describe('BoxplotDirective', () => {
+  it('should render no data placeholder when no data is passed', async () => {
+    await renderTestComponent({ points: [] });
+    expect(
+      screen.getByLabelText('No data is currently available.')
+    ).toBeVisible();
+  });
```

Comment:
The chart is rendered as a canvas element, so I used the `aria-label` generated by [Apache ECharts](https://echarts.apache.org/handbook/en/best-practices/aria/) to confirm that the correct chart has been displayed. 

I initially wanted to add [Playwright component testing](https://playwright.dev/docs/test-components) to use screenshots for visual comparison / testing, but component testing is not available for Angular yet and is still an experimental feature for other frameworks (e.g. React).